src/js/types/trezor/protobuf.js

```diff
@@ -811,6 +818,18 @@ export type CardanoTxMint = {
     asset_groups_count: number,
 };
 
+// CardanoTxCollateralInput
+export type CardanoTxCollateralInput = {
+    prev_hash: string,
+    prev_index: number,
+};
+
+// CardanoTxRequiredSigner
+export type CardanoTxRequiredSigner = {
+    key_hash?: string,
+    key_path: number[],
```

Comment:
I think the `key_path` should be optional. So it needs to be added to `RULE_PATCH` in `scripts/protobuf-patches.js`.
```suggestion
    key_path?: number[],
```

Reply:
Fixed. But it's weird that we were sending `CardanoTxRequiredSigner.key_path: undefined` in some cases (see `CardanoSignTransaction.js`) even before the fix and `yarn run flow` didn't complain.

Reply:
Maybe it's because each part of the `payload.requiredSigners.map` expression is of type `any` and flow just gives up. And maybe it's intentionally setup that way so we wouldn't have to cast every transformation. 🤷 

Actually I just noticed that all the other transformations have definite types. So I think we need to add a cast to the required signers:
```javascript
return ({
  key_path: requiredSigner.keyPath
      ? validatePath(requiredSigner.keyPath, 3)
      : undefined,
  key_hash: requiredSigner.keyHash,
  }: CardanoTxRequiredSigner)
```
When the cast is added, flow does complain about the undefined.
  
And then we should also add a return type to `transformInput` and `transformCollateralInput` functions - sorry I missed it before. In `transformOutput` the return type can be inferred so it doesn't have to be explicitly added, but that apparently isn't the case for the `transformInput` functions.

Reply:
:+1: Added.